docs/dev/relay_pass_infra.rst

```diff
@@ -0,0 +1,616 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Relay Pass Infra
+==================================
+
+Relay features a series of optimization passes which improve performance metrics
+of models such as mean inference, memory footprint, or power consumption for
+specific devices. There are a suite of standard optimizations as well as machine
+learning specific optimizations including constant folding, dead code
+elimination, operator layout alteration, and operator fusion, etc. Each of these
+passes are structured as Relay-to-Relay transformations on the abstract syntax
+tree (AST) using the analysis result collected during and/or before traversal.
+
+On one hand, many existing production compilers, such as GCC and LLVM, employ
```

Comment:
It is confusing to have two paragraphs start with "On one hand" and "on the other" without explicitly setting up the contrast/tension beforehand. That is, there should be a sentence or paragraph first that sets up the tension between these two approaches.